winch/codegen/src/isa/aarch64/asm.rs

```diff
@@ -474,15 +474,54 @@ impl Assembler {
         self.emit(Inst::Ret {});
     }
 
+    /// An unconditional branch.
+    pub fn jmp(&mut self, target: MachLabel) {
+        self.emit(Inst::Jump {
+            dest: BranchTarget::Label(target),
+        });
+    }
+
+    /// A conditional branch.
+    pub fn jmp_if(&mut self, kind: Cond, taken: MachLabel) {
+        self.emit(Inst::CondBr {
+            taken: BranchTarget::Label(taken),
+            not_taken: BranchTarget::ResolvedOffset(4),
+            kind: CondBrKind::Cond(kind),
+        });
+    }
+
+    /// Emits a jump table sequence.
+    pub fn jmp_table(
+        &mut self,
+        targets: &[MachLabel],
+        default: MachLabel,
+        index: Reg,
+        tmp1: Reg,
+        tmp2: Reg,
+    ) {
+        self.emit(Inst::JTSequence {
```

Comment:
We'll need an `EmitIsland` instruction here too; see [here](https://github.com/bytecodealliance/wasmtime/blob/ba864e987ef1ab87c439ca6b396264547d2425e1/cranelift/codegen/src/isa/aarch64/lower.isle#L2967) and [here](https://github.com/bytecodealliance/wasmtime/blob/ba864e987ef1ab87c439ca6b396264547d2425e1/cranelift/codegen/src/isa/aarch64/inst.isle#L933-L952) for more details on how and why.